src/bot/composers/messages/denylist.composer.ts

```diff
@@ -0,0 +1,59 @@
+import escapeHTML from 'escape-html';
+import { Composer } from 'grammy';
+
+import { LOGS_CHAT_THREAD_IDS } from '../../../const';
+import { logsChat } from '../../../creator';
+import { urlLogsStartMessage } from '../../../message';
+import { getDeleteDenylistMessage } from '../../../message/denylist.message';
+import type { GrammyContext } from '../../../types';
+import { getUserData, telegramUtil } from '../../../utils';
+
+/**
+ * @description Remove strategic information logic
+ */
+export const getDenylistComposer = () => {
+  const denylistComposer = new Composer<GrammyContext>();
+
+  /**
+   * @param {GrammyContext} context
+   * @param {string} [message]
+   * */
+  async function logDenylistMessage(context: GrammyContext, message?: string) {
+    const { userMention, chatMention } = await telegramUtil.getLogsSaveMessageParts(context);
+    const text = message || context.state?.text || '';
+    const { denylist } = context.chatSession.chatSettings;
+    return context.api.sendMessage(
+      logsChat,
+      `${urlLogsStartMessage} (${denylist?.join(', ')}) by user ${userMention}:\n\n${chatMention || userMention}\n${escapeHTML(text)}`,
+      {
+        parse_mode: 'HTML',
+        message_thread_id: LOGS_CHAT_THREAD_IDS.DENYLIST,
+      },
+    );
+  }
+
+  /**
+   * Middleware to handle denylist checks and message deletion.
+   */
+  denylistComposer.use(async (context, next) => {
+    const { text } = context.state;
+    const { chatSettings } = context.chatSession;
+    const { denylist } = chatSettings;
+
+    if (Array.isArray(denylist) && denylist.length > 0 && text) {
+      const denyWord = denylist.find((word) => text.toLowerCase().includes(word.toLowerCase()));
+      if (denyWord) {
+        await context.deleteMessage();
+        await logDenylistMessage(context, text);
```

Comment:
Could you please log the exception word instead of the entire list of words? Logging all the words could break our logs, and we don't really need to. It's more important for us to know the word than the list.